### PR TITLE
Fixed several docblocks to match typehints

### PR DESCRIPTION
**Pull request by weirdan**

### CODE REVIEW

File: src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ public function __construct(Parser $parser, ?ErrorHandler $errorHandler)
     /**
      * @param Traversable $files
      *
-     * @return Traversable|array[] a series of AST roots, one for each given file
+     * @return Traversable a series of AST roots, one for each given file
      */
     public function __invoke(Traversable $files): Traversable
```
Comment (Ocramius) [inline src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php]: We can probably switch to `iterable` /cc @maglnet

Comment (Ocramius) [reply src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php]: @weirdan the point of `array[]` is to indicate that each element of `$files` is an `array` - the change is incorrect here

File: src/ComposerRequireChecker/DefinedSymbolsLocator/LocateDefinedSymbolsFromASTRoots.php [hunk 2]
```diff
@@ -10,7 +10,7 @@
 final class LocateDefinedSymbolsFromASTRoots
 {
     /**
-     * @param Traversable|array[] $ASTs a series of AST roots
+     * @param Traversable $ASTs a series of AST roots
```
Comment (Ocramius) [inline src/ComposerRequireChecker/DefinedSymbolsLocator/LocateDefinedSymbolsFromASTRoots.php]: Same as above:

@weirdan the point of `array[]` is to indicate that each element of `$files` is an `array` - the change is incorrect here

File: src/ComposerRequireChecker/UsedSymbolsLocator/LocateUsedSymbolsFromASTRoots.php [hunk 3]
```diff
@@ -10,7 +10,7 @@
 final class LocateUsedSymbolsFromASTRoots
 {
     /**
-     * @param Traversable|array[] $ASTs a series of AST roots
+     * @param Traversable $ASTs a series of AST roots
```
Comment (Ocramius) [inline src/ComposerRequireChecker/UsedSymbolsLocator/LocateUsedSymbolsFromASTRoots.php]: Same as above:

@weirdan the point of `array[]` is to indicate that each element of `$files` is an `array` - the change is incorrect here

File: src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php (hunk 1 above)
Comment (weirdan) [reply src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php]: oh, so that's phpstorm-style generics then, and not a union type? Ok.

Comment (Ocramius) [reply src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php]: `array[]` is `array<array>`, basically

Comment (weirdan) [reply src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php]: It's not an array though, as it's used in `file_get_contents()` (thus it's a string). So as far as I understand this should be `Traversable|string[]` in this case (based on [this comment](https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000143944-PHPDoc-collection-generics-notation-supported-?page=1#community_comment_115000171490))

Comment (Ocramius) [reply src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php]: @weirdan needs adjustment then, but not removal :+1:

**Comment by weirdan** [comment]
@Ocramius done.

**Comment by Ocramius** [comment]
Thanks @weirdan!

**Pull request merged**